File: src/ShoppingCart.API/Campaign/CreateCampaignRequest.cs
using System;
using ShoppingCart.Business.Campaign;

namespace ShoppingCart.API.Campaign
{
    public class CreateCampaignRequest
    {
        public Guid CategoryID { get; set; }

        public int MinimumItemCount { get; set; }

        public DiscountType Type { get; set; }

        public decimal Rate { get; set; }
    }
}

File: spring-boot/src/main/java/com/erdaldalkiran/cart/api/campaign/CreateCampaignRequest.java
package com.erdaldalkiran.cart.api.campaign;

import com.erdaldalkiran.cart.application.discount.OnlyValidDiscountTypes;
import com.erdaldalkiran.cart.domain.discount.DiscountType;
import lombok.Getter;
import lombok.Setter;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Positive;
import java.math.BigDecimal;
import java.util.UUID;

@Getter
@Setter
public class CreateCampaignRequest {
    @NotNull
    private UUID categoryID;

    @Positive
    private int minimumItemCount;

    @NotNull
    @OnlyValidDiscountTypes
    private DiscountType discountType;

    @NotNull
    @Positive
    private BigDecimal rate;
}
